crates/subspace-farmer/src/bin/subspace-farmer/commands/farm.rs

```diff
@@ -853,28 +854,35 @@ where
     let networking_fut = networking_fut;
     let farm_fut = farm_fut;
     let farmer_cache_worker_fut = farmer_cache_worker_fut;
+    let prometheus_worker = prometheus_worker;
 
     let networking_fut = pin!(networking_fut);
     let farm_fut = pin!(farm_fut);
     let farmer_cache_worker_fut = pin!(farmer_cache_worker_fut);
 
     select! {
         // Signal future
-        _ = signal.fuse() => {}
+        () = signal.fuse() => {}
 
         // Networking future
-        _ = networking_fut.fuse() => {
-            info!("Node runner exited.")
+        networking_error = networking_fut.fuse() => {
+            info!(?networking_error, "Node runner exited.")
         },
 
         // Farm future
         result = farm_fut.fuse() => {
+            info!("Farm exited.");
             result??;
         },
 
         // Piece cache worker future
-        _ = farmer_cache_worker_fut.fuse() => {
-            info!("Farmer cache worker exited.")
+        farmer_cache_worker_error = farmer_cache_worker_fut.fuse() => {
+            info!(?farmer_cache_worker_error, "Farmer cache worker exited.")
```

Comment:
Exit reasons are identical to networking stack, I don't think it is necessary to log anything extra here (and it is a result, not an error).